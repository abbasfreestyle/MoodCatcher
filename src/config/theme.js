const palette = {
  charcoal: '#393E41',
  trout: '#4A4C58',
  darkGreen: '#44BBA4',
  calmBlue: '#81DBD1',
  brightGreen: '#F2FCFB',
  grey: '#EEEEEE',
  white: '#FFFFFF'
};

const theme = {
  primary: palette.darkGreen,
  primaryVariant: palette.calmBlue,
  secondary: palette.brightGreen,
  surface: palette.white,

  onPrimary: palette.white,
  onPrimaryVariant: palette.white,
  onSecondary: palette.charcoal,
  onSurface: palette.trout,

  shadow: palette.darkGreen,
  disabled: palette.grey
};

export default theme;
