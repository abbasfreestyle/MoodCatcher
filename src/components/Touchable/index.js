import React from 'react';
import PropTypes from 'prop-types';

import {
  View,
  Platform,
  StyleSheet,
  TouchableNativeFeedback,
  TouchableWithoutFeedback
} from 'react-native';

import theme from 'theme';

const isIos = Platform.OS === 'ios';

const styles = StyleSheet.create({
  overflow: {
    overflow: 'hidden'
  }
});

const Touchable = ({ children, borderRadius, rippleColor, ...props }) => {
  const Button =
    isIos || Platform.Version < 21
      ? TouchableWithoutFeedback
      : TouchableNativeFeedback;
  const androidRipple = !isIos
    ? {
        background: TouchableNativeFeedback.Ripple(rippleColor),
        useForeground: true
      }
    : undefined;
  return (
    <Button {...androidRipple} {...props}>
      <View style={[styles.overflow, { borderRadius }]}>{children}</View>
    </Button>
  );
};

Touchable.defaultProps = {
  borderRadius: 0,
  rippleColor: theme.primary
};

Touchable.propTypes = {
  children: PropTypes.node.isRequired,
  borderRadius: PropTypes.number,
  rippleColor: PropTypes.string
};

export default Touchable;
