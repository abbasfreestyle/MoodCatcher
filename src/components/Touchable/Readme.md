# Touchable

The Touchable component extends off React Native's [TouchableNativeFeedback](https://facebook.github.io/react-native/docs/touchablenativefeedback) and [TouchableWithoutFeedback](https://facebook.github.io/react-native/docs/touchablewithoutfeedback). It's a flexible component that accepts a single element as a child prop. The purpose for this is so that the touchable component is `Platform specific`.

- If it's an Android device, it utilises the ripple effect.
- If it's iOS, it defaults to a tappable area with `no feedback`.

## Android

The android version has a ripple effect on tap. This is supported with Android API 21+. Versions that don't support this will output a yellow warning but not affect the functionality of the button.

The ripple color accepts the `rippleColor` prop. By default the ripple color is set to `theme.onSurface` in [theme](../../../config/theme/index.js)

## Border Radius

Apply the `borderRadius` prop to Touchable `instead of the styled View`. The reason for this is because it prevents the android ripple effect from spilling out.

## Usage

```js
import { Touchable } from 'components';

class MyButton extends React.Component {
  render() {
    return (
      <Touchable onPress={() => console.log('hi')} borderRadius={10}>
        <View styles={styles.myButtonStyle}>
          <SomeIcon />
          <Text>This is a button</Text>
        </View>
      </Touchable>
    );
  }
}
```

Also see the [LoginButton](../LoginButton/index.js) to see how the Touchable component is used there.