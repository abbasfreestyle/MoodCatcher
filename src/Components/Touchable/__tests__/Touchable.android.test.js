import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import theme from 'theme';
import Touchable from '..';

jest.mock('Platform', () => ({ OS: 'android' }));

const requiredProps = {
  children: <Text>Test Button</Text>
};

const wrapper = shallow(<Touchable {...requiredProps} />);

describe('Touchable Android.', () => {
  describe('Snapshots.', () => {
    beforeEach(() => {
      wrapper.setProps({ ...Touchable.defaultProps, ...requiredProps });
    });

    test('should match default props.', () => {
      expect(Touchable.defaultProps).toMatchSnapshot();
    });

    test('should match with default props provided.', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when border radius is changed', () => {
      wrapper.setProps({ borderRadius: 50 });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('should match when android rippleColor is changed', () => {
      wrapper.setProps({ rippleColor: theme.onPrimary });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
