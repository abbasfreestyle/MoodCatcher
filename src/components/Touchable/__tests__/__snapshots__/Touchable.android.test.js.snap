// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable Android. Snapshots. should match default props. 1`] = `
Object {
  "borderRadius": 0,
  "rippleColor": "#44BBA4",
}
`;

exports[`Touchable Android. Snapshots. should match when android rippleColor is changed 1`] = `
<DummyTouchableNativeFeedback
  background={Object {}}
  useForeground={true}
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  >
    <Text>
      Test Button
    </Text>
  </View>
</DummyTouchableNativeFeedback>
`;

exports[`Touchable Android. Snapshots. should match when border radius is changed 1`] = `
<DummyTouchableNativeFeedback
  background={Object {}}
  useForeground={true}
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 50,
        },
      ]
    }
  >
    <Text>
      Test Button
    </Text>
  </View>
</DummyTouchableNativeFeedback>
`;

exports[`Touchable Android. Snapshots. should match with default props provided. 1`] = `
<DummyTouchableNativeFeedback
  background={Object {}}
  useForeground={true}
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  >
    <Text>
      Test Button
    </Text>
  </View>
</DummyTouchableNativeFeedback>
`;
